dataland-internal-storage/src/main/kotlin/db/migration/V6__FlattenEnvironmentalObjectiveMapsInEuTaxonomyForNonFinancials.kt

```diff
@@ -0,0 +1,88 @@
+import org.json.JSONArray
+import org.json.JSONObject
+
+/**
+ * This migration script updates the old version eutaxonomy for non financials datasets to the new version
+ * and the new version is integrated into the old datatype
+ */
+class V6__FlattenEnvironmentalObjectiveMapsInEuTaxonomyForNonFinancials : BaseJavaMigration() {
+    private enum class CriteriaType(val prefix: String) {
+        SubstantialContribution("substantialContribution"),
+        Dnsh("dnsh"),
+    }
+
+    override fun migrate(context: Context?) {
+        migrateCompanyAssociatedDataOfDatatype(
+            context,
+            "eutaxonomy-non-financials",
+            this::flattenEnvironmentalObjectiveMaps,
+        )
+    }
+
+    /**
+     * Migrates an old eu taxonomy non financials dataset to the new format
+     */
+    fun flattenEnvironmentalObjectiveMaps(dataTableEntity: DataTableEntity) {
+        val dataset = dataTableEntity.dataJsonObject
+        listOf("revenue", "capex", "opex").forEach {
+            migrateCashFlow(dataset.getOrJavaNull(it) as JSONObject? ?: return@forEach)
+        }
+        dataTableEntity.companyAssociatedData.put("data", dataset.toString())
+    }
+
+    private fun migrateCashFlow(cashFlow: JSONObject) {
+        migrateFieldNames(cashFlow)
+        migrateMapForCriteria(cashFlow, CriteriaType.SubstantialContribution)
+        (cashFlow.getOrJavaNull("alignedActivities") as JSONArray? ?: return).forEach {
+            migrateMapForCriteria(it as JSONObject, CriteriaType.SubstantialContribution)
+            migrateMapForCriteria(it, CriteriaType.Dnsh)
+        }
+    }
+
+    private fun migrateMapForCriteria(baseObject: JSONObject, criteriaType: CriteriaType) {
+        val fieldRenamingHelper = mapOf(
+            "ClimateMitigation" to "ClimateChangeMitigation",
+            "ClimateAdaptation" to "ClimateChangeAdaption",
+            "Water" to "SustainableUseAndProtectionOfWaterAndMarineResources",
+            "CircularEconomy" to "TransitionToACircularEconomy",
+            "PollutionPrevention" to "PollutionPreventionAndControl",
+            "Biodiversity" to "ProtectionAndRestorationOfBiodiversityAndEcosystems",
+        )
+        val criteriaObject = baseObject.opt("${criteriaType.prefix}Criteria") ?: return
+        if (criteriaObject is JSONObject) {
+            fieldRenamingHelper.keys.forEach { oldFieldName ->
+                if (criteriaObject.has(oldFieldName)) {
+                    baseObject.put(
+                        "${criteriaType.prefix}To${fieldRenamingHelper[oldFieldName]}",
+                        criteriaObject.getOrJsonNull(oldFieldName),
+                    )
+                }
+            }
+        }
+        baseObject.remove("${criteriaType.prefix}Criteria")
+    }
+
+    private fun migrateFieldNames(baseObject: JSONObject) {
+        val fieldsToRename = listOf(
+            "nonEligibleShare", "eligibleShare", "nonAlignedShare", "alignedShare", "enablingShare", "transitionShare"
```

Comment:
It should be transitionalShare, not transitionShare. Maye that fixes the error

Reply:
I verfified it for the company ID 69154e5c-2408-4aed-938a-541c8055f488. After i manually changed the totalTransitionalShare to transitionalShare the dataset could be retrieved. So after changing this, the migration should work as intended.